src/cdk/scrolling/scrolling.md

```diff
@@ -56,6 +56,9 @@ is reused instead. The size of the view cache can be adjusted via the `templateC
 property; setting this size to `0` disables caching. If your templates are expensive in terms of
 memory you may wish to reduce this number to avoid spending too much memory on the template cache.
 
+Recycled components within the `<cdk-virtual-scroll-viewport>` preserve state like bound parameter values. 
+When rendering dynamic components, you may want to disable caching completely.
```

Comment:
Thanks for adding to the docs. Based on your initial issue it looks like a lot of people would benefit from this sort of messaging. However, I don't think "dynamic components" will be clear enough. Is there another way to describe what users should watch out for in more Angular-ish terms? Should users specifically be careful about components not calling their constructor or ngOnInit when reused?